@W-19809645 backport slas proxy callsback from PR #3411

clavery opened the pull request:
backport slas proxy callsback from PR #3411

See that PR for details of the functionality

clavery committed:
backport slas proxy callsback from PR #3411

backport slas proxy callsback from PR #3411

slas private proxy test coverage

lint

lint warnings

backporting SLAS proxy callbacks

Remove passwordless login 404 masking (not part of PR #3411)

Restore error logging from base branch (preserve existing functionality)

clavery committed:
updating changelog

cc-prodsec commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/ffc30ecf-7af5-4f61-9b76-c23c2f9072f7/license?refs=bfbf203acc3af4e97ee8b85455a39196cdb2bcad&source=prChecksComment) |
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/ffc30ecf-7af5-4f61-9b76-c23c2f9072f7?refs=bfbf203acc3af4e97ee8b85455a39196cdb2bcad&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

clavery commented:
@shethj This is the backport. CI seems to be failing on an outdated lockfile (but we've made not package changes here). Seems like I might need a fresh lockfile to get it to pass

clavery committed:
Add selfHandleResponse and remove path validation tests

- Add selfHandleResponse: true required for responseInterceptor
- Remove tests for 403 path validation (not in 3.10.x branch)

clavery committed:
non-function change to sdk

alexvuong commented on packages/commerce-sdk-react/CHANGELOG.md:
```diff
@@ -1,3 +1,5 @@
+## v3.10.1 (Oct 27, 2025)
```
This does not seem right? 🤔

clavery replied:
The tests were failing because the lockfiles were out of date on that package in this branch. Then, because the package has a change we needed a changelog update. I saw in other packages we've done empty changelog

I can also skip it or say "updated lockfiles"

alexvuong replied:
I think you can update the lockfiles, and skip the changelog. If you want to update changlog, the correct is 3.3.1 since commerce-react-sdk version is independent from the general pwa-kit sdk
https://github.com/SalesforceCommerceCloud/pwa-kit/releases/tag/v3.10.0

clavery replied:
Cool thanks. Let me do that and get CI passing (and I need to manually test the feature again)

alexvuong commented on packages/pwa-kit-runtime/package-lock.json:
```diff
@@ -51,7 +51,6 @@
       "version": "2.3.0",
```
This lock file still has 3.10.0, are they supposed to be 3.10.1?

clavery replied:
Yeah I'll bump it

clavery commented:
FYI holding this in draft form to see if the SCAPI change can be consumed by the customer rather than needing a backport.
